pandas/core/indexes/base.py

```diff
@@ -3122,23 +3125,67 @@ def isin(self, values, level=None):
 
             .. versionadded:: 0.18.1
 
-            Support for values as a set
+            Support for values as a set.
 
-        level : str or int, optional
+        level : str or int, optional in the case of Index, compulsory on
```

Comment:
Not quite, it's always optional but we don't handle it properly see https://github.com/pandas-dev/pandas/issues/20252